tasks/omnibus.py

```diff
@@ -379,3 +386,46 @@ def manifest(
         omnibus_s3_cache=False,
         log_level=log_level,
     )
+
+
+@task
+def rpath_edit(ctx, install_path, target_rpath_dd_folder, platform="linux"):
+    # Since bash and sh don't support **/* we are using glob
+    # to get all datadog files inside of one `file --mime-type` command
+    datadog_files = glob.glob(f"{install_path}/**/*", recursive=True)
+    with open("/tmp/ddargs", "w") as f:
+        f.write(" ".join(datadog_files))
+    files = ctx.run("cat /tmp/ddargs | xargs file --mime-type", hide=True).stdout
+
+    for line in files.split('\n'):
+        if not line:
+            continue
+        file, file_type = line.split(":")
+        file_type = file_type.strip()
+
+        if platform == "linux":
+            if file_type not in ["application/x-executable", "inode/symlink", "application/x-sharedlib"]:
+                continue
+            binary_rpath = ctx.run(f'objdump -x {file} | grep "RPATH"', warn=True, hide=True).stdout
+        else:
+            if file_type != "application/x-mach-binary":
+                continue
+            binary_rpath = ctx.run(f'otool -l {file} | grep -A 2 "RPATH"', warn=True, hide=True).stdout
+
+        if install_path in binary_rpath:
+            new_rpath = os.path.relpath(target_rpath_dd_folder, os.path.dirname(file))
+            if platform == "linux":
+                ctx.run(f"patchelf --force-rpath --set-rpath \\$ORIGIN/{new_rpath}/embedded/lib {file}")
+            else:
+                # The macOS agent binary has 18 RPATH definition, replacing the first one should be enough
+                # but just in case we're replacing them all.
+                # We're also avoiding unnecessary `install_name_tool` call as much as possible.
+                exit_code = 0
+                rpath_line_number = binary_rpath.count('\n') // 3
+                while rpath_line_number > 0 and exit_code == 0:
+                    exit_code = ctx.run(
+                        f"install_name_tool -rpath {install_path}/embedded/lib @loader_path/{new_rpath}/embedded/lib {file}",
+                        warn=True,
+                        hide=True,
+                    ).exited
+                    rpath_line_number -= 1
```

Comment:
How much does the extra logic around `rpath_line_number` save in terms of execution time? I find it somewhat clunky (partly in how it's coupled to the fact that it depends on the `-A 2` argument used to call grep.

Maybe we should just consider replacing it just once if that works (as it should) and don't worry too much about other RPATHs ending up in the binary (at least for now).

At any rate, here goes a suggestion based on the current logic:
```suggestion
                number_of_rpaths = binary_rpath.count('\n') // 3
                for _ in range(number_of_rpaths):
                    exit_code = ctx.run(
                        f"install_name_tool -rpath {install_path}/embedded/lib @loader_path/{new_rpath}/embedded/lib {file}",
                        warn=True,
                        hide=True,
                    ).exited
                    if exit_code != 0:
                        break
```

Reply:
I'd prefer replace too much than not enough to be safe especially for the first iteration. If unnecessary we can always remove the logic later. 👍 for the suggestion